nuxeo-features/nuxeo-platform-tag-service/nuxeo-platform-tag-core/src/main/java/org/nuxeo/ecm/platform/tag/RelationTagService.java

```diff
@@ -82,7 +82,8 @@ public void doTag(CoreSession session, String docId, String label) {
 
     @Override
     public void doUntag(CoreSession session, String docId, String label) {
-        if (session.getDocument(new IdRef(docId)).isProxy()) { // Tags are disabled on proxies
+        IdRef ref = new IdRef(docId);
+        if (session.exists(ref) && session.getDocument(ref).isProxy()) { // Tags are disabled on proxies
```

Comment:
Is the `exists` on purpose here? Why not just fail with the regular DocumentNotFoundException if the doc does not exist?

Reply:
The `exists` is here because we can have the case of untagging when we delete a document in the `TaggedVersionListener`. So before checking if it's a proxy for regular untagging, we need to check that the document exists before.